dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/sharing/Sharing.java

```diff
@@ -211,4 +220,37 @@ public Sharing copy()
             .users( new HashMap<>( users ) )
             .userGroups( new HashMap<>( userGroups ) ).build();
     }
+
+    public Sharing withAccess( UnaryOperator<String> accessTransformation )
+    {
+        return builder()
+            .external( external )
+            .publicAccess( accessTransformation.apply( publicAccess ) )
+            .owner( owner )
+            .users( mapValues( users,
+                user -> new UserAccess( accessTransformation.apply( user.getAccess() ), user.getId() ) ) )
+            .userGroups( mapValues( userGroups,
+                group -> new UserGroupAccess( accessTransformation.apply( group.getAccess() ), group.getId() ) ) )
+            .build();
+    }
+
+    private static <K, V> Map<K, V> mapValues( Map<K, V> map, UnaryOperator<V> mapper )
+    {
+        if ( map == null )
+        {
+            return null;
+        }
+        return map.entrySet().stream()
+            .collect( toMap( Entry::getKey, e -> mapper.apply( e.getValue() ) ) );
+    }
+
```

Comment:
Could be nice with an example of how the access string looks like before/after to better understand the operation.

Reply:
Added. See below.